src/components/commit-list/CommitList.js

```diff
@@ -27,25 +27,48 @@ const Commit = ({ sha, commit, commiter, author, url }) => {
   );
 };
 
-export const CommitList = ({ commits }) => {
+export const CommitList = ({ commits, branches }) => {
+  const onSelect = (event) => {
+    // TODO: selecting a branch should reload the page to that branch
+    event.preventDefault();
+  };
+
+  const SelectDropdown = () => (
+    <Flex justifyContent="flex-end" mb="12px">
+      <Select fontSize="15px" onChange={onSelect} w="160px">
+        {branches.map((branchName, index) => {
+          return (
+            <option key={index} value={branchName}>
+              {branchName}
+            </option>
+          );
+        })}
+      </Select>
+    </Flex>
+  );
+
   return commits.length ? (
-    <Box mt="40px" textAlign="left">
-      {commits.map(({ author, commit, commiter, html_url, sha }) => (
-        <Commit
-          key={sha}
-          sha={sha}
-          commit={commit}
-          url={html_url}
-          commiter={commiter}
-          author={author}
-        />
-      ))}
-    </Box>
+    <>
+      <Box mt="40px" textAlign="left">
+        <SelectDropdown />
+        {commits.map(({ author, commit, commiter, html_url, sha }) => (
```

Comment:
nit - We could pass the whole `commit` to the Commit component and just destructure once in Commit which would save space